segmentation_models_pytorch/losses/_functional.py

```diff
@@ -196,15 +182,28 @@ def soft_tversky_score(
     eps: float = 1e-7,
     dims=None,
 ) -> torch.Tensor:
+    """Tversky loss
+
+    References:
+        https://arxiv.org/pdf/2302.05666
+        https://arxiv.org/pdf/2303.16296
+
+    """
     assert output.size() == target.size()
     if dims is not None:
-        intersection = torch.sum(output * target, dim=dims)  # TP
-        fp = torch.sum(output * (1.0 - target), dim=dims)
-        fn = torch.sum((1 - output) * target, dim=dims)
+        difference = torch.norm(output - target, p=1, dim=dims)
+        output_sum = torch.sum(output, dim=dims)
+        target_sum = torch.sum(target, dim=dims)
+        intersection = (output_sum + target_sum - difference) / 2  # TP
+        fp = output_sum - intersection
+        fn = target_sum - intersection
```

Comment:
Hi @qubvel, thanks for reviewing!

Since `dim=None` are default in `torch.norm` and `torch.sum`, should we just remove the if/else?

Reply:
Sounds good 👍

Reply:
I've removed the if/else, and changed `torch.norm` in my previous commit to `torch.linalg.vector_norm`, since `torch.norm` is [deprecated](https://pytorch.org/docs/stable/generated/torch.norm.html#) and may be removed in a future PyTorch release.